### ISSUE TITLE
LED Model Number and Web UI Power/HDD LED's not working

### ISSUE THREAD

**Issue by Johannp02180** [zappanaut/usb-atx-ctrl#3]
Hello!

I was able to sucessfully build it but the LED's aren't working so I wanted to see which LED's you used in yours. Also I setup my override file very similar to yours but the icons for the power and HDD aren't working, do you know what I might be doing wrong? Are they related to the LED's not lighting up?

**Comment by zappanaut** [comment]
Hi!

Sorry, that the LEDs are not working (yet). The "LEDs" in the web application are 1:1 "wired" to the LEDs on the board.
The type of the LED only has an influence on the brightness. But even if you don't put any they should work in the web application. There may be several reasons if it does not work at all:

1. The polarity of the LED connector from the mainboard is wrong.
2. The PhotoMOS relay is soldered the wrong way, or is broken (unlikely).
3. The impedance of the mainboard's LED voltage is to high to trigger the PhotoMOS relay.

(1.) and (2.) should be easy to fix or check. Next you can test if it works in principle by applying 3.3V or 5V from the RP2040 board to the LED inputs. If this works it is most probably a problem with the impedance and you have to lower the resistors R3 and R4. Around 330 Ohm should be a safe bet. Maybe you  measure the voltage across the power LED input on the board?

Regards
Peter

**Comment by Johannp02180** [comment]
I'm not close to the server to test the components but I attached a photo of the PCB I soldered. I think something I could possible fix remotely is the web application? I attached a photo of what I see in the web application as well.

Thank you!
![IMG_2494](https://github.com/zappanaut/usb-atx-ctrl/assets/11222092/7634802c-e0bd-401a-99a5-833ff66c6624)


<img width="504" alt="Screenshot 2023-07-16 at 12 18 45 AM" src="https://github.com/zappanaut/usb-atx-ctrl/assets/11222092/460ebcf6-a6ef-452a-8acd-606f7ff44613">

**Comment by zappanaut** [comment]
The PCB looks good. I assume "power" and "reset" buttons are working, no? Did you flash the firmware from here: https://github.com/zappanaut/pico-usb-io-board?
For now my best guess is that it is a wiring (polarity) problem or the input resistor value for the PhotoMOS diodes is too high for your mainboard.
There is nothing to fix in the web application if the signal is not there. You can check the pin assignments in your override.yaml again, but I fear further debugging has to wait until you have access to the server again.

**Comment by Johannp02180** [comment]
Yes, the power and reset buttons are working. The buttons in the web application also work. I used the same resistor values referenced in your BOM, but I'll double-check the values I used. Thank you!

**Comment by zappanaut** [comment]
Ok, so it seems 560 Ohm for R3 and R4 is a bit too high for your mainboard. These values just worked for me and also kept the amount of resistor values in the BOM low.
If you don't have lower value resistors you can also go with 280 Ohm by simply soldering a second 560 Ohm resistor on top of the old one.

**Comment by Johannp02180** [comment]
Ok, that makes sense I just assumed the LED's I used were the same without bothering to check the resistor values and LED voltages (oopsies). However, the web application should work otherwise. I'll have to double-check my connections and test if there is a signal on those pins. Thank you.

**Comment by zappanaut** [comment]
Just to be a bit more clear ... the problem might be with the input resistors for the PhotoMOS (R3 and R4). The current limiting resistors for the LEDs does not effect the functionality but only the brightness.

**Comment by Johannp02180** [comment]
Got it, I used 560ohm for R3 and R4.

**Comment by Johannp02180** [comment]
I haven't had the chance to go back and test the components you mentioned but the power on/off does work, (really well!). If I wanted to keep the functionality of my existing power button, if I'm not mistaken are the existing button/front panel headers connections just parallel to the ones on your board?

**Comment by zappanaut** [comment]
Yes, you can connect the front panel connectors in parallel to the board. I used ribbon cables to connect the PCB to the mainboard and simply crimped such an IDC connector to the cable: https://m.media-amazon.com/images/I/71Lu1ZdaqwL._AC_SL1500_.jpg

**Comment by andrewrabert** [comment]
For those curious, I'm successfully using these LEDs on my builds:
- [Dialight 599-0480-127F (green)](https://www.mouser.com/ProductDetail/645-599-0480-127F)
- [Dialight 599-0411-127F (red)](https://www.mouser.com/ProductDetail/645-599-0411-127F)
